synapse/cryotank.py

```diff
@@ -320,12 +318,20 @@ def _onCryoSlice(self, chan, mesg):
         offs = mesg[1].get('offs')
         size = mesg[1].get('size')
 
-        tank = self.tanks.get(name)
-        if tank is None:
-            return chan.txfini()
+        with chan:
+
+            tank = self.tanks.get(name)
+            if tank is None:
+                return chan.tx((False, ('NoSuchName', {'name': name})))
 
-        #TODO: make this a chunked yielder
-        chan.txfini(tuple(tank.slice(offs, size)))
+            chan.setq()
+            chan.tx((True, True))
+
+            genr = tank.slice(offs, size)
+            genr = s_common.chunks(genr, 100)
+
+            # 100 chunks of 100 in flight...
+            chan.txwind(genr, 100, timeout=30)
```

Comment:
General comment for all of these. Do we want the chunksize, windowsize, timeout, etc to be configurable?